### PR TITLE
fix(core): ensure page.title is string, close #1306

### LINKED ISSUE #1306: [Bug report] Page.title is not type guarded
### Description

```md
---
title: 1
---
```

```md
---
title: {}
```

There will be noob users who set titles like this, and `page.title` (with `string` type) will have other types, and this may break the application.

Unless we set `page.title: unknown` , we should guard the types of it.

### Reproduction

Not needed

### Used Package Manager

npm

### System Info

```shell
.
```

### PR DESCRIPTION
**Pull request by Mister-Hope**

### CODE REVIEW

File: packages/core/src/page/renderPageContent.ts [hunk 1]
```diff
@@ -74,6 +74,6 @@ export const renderPageContent = ({
       'frontmatter'
     ),
     sfcBlocks,
-    title: frontmatter.title ?? title,
+    title: isString(frontmatter.title) ? frontmatter.title : title,
```
Comment (meteorlxy) [inline packages/core/src/page/renderPageContent.ts:77]: Maybe

```
frontmatter.title ? `${frontmatter.title}` : title
```

Comment (Mister-Hope) [reply packages/core/src/page/renderPageContent.ts:77]: Emm, then why not `String(fromtmatter.title)`

Comment (meteorlxy) [reply packages/core/src/page/renderPageContent.ts:77]: As I remember, string template has better performance than string constructor in most cases.

The only difference is that string constructor can handle symbol, while string template cannot. But I think frontmatter.title is impossible to be a symbol value in this case.

Comment (Mister-Hope) [inline packages/core/src/page/renderPageContent.ts:77]:
```suggestion
    title: frontmatter.title ? `${frontmatter.title}` : title,
```

**Comment by Mister-Hope** [comment]
This should be ready

**Pull request merged**